x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/Connector.java

```diff
@@ -172,25 +185,179 @@ public Connector(StreamInput in) throws IOException {
         this.syncNow = in.readBoolean();
     }
 
-    private static final ParseField ID_FIELD = new ParseField("connector_id");
-    private static final ParseField API_KEY_ID_FIELD = new ParseField("api_key_id");
-    private static final ParseField CONFIGURATION_FIELD = new ParseField("configuration");
-    private static final ParseField CUSTOM_SCHEDULING_FIELD = new ParseField("custom_scheduling");
-    private static final ParseField DESCRIPTION_FIELD = new ParseField("description");
-    private static final ParseField ERROR_FIELD = new ParseField("error");
-    private static final ParseField FEATURES_FIELD = new ParseField("features");
-    private static final ParseField FILTERING_FIELD = new ParseField("filtering");
-    private static final ParseField INDEX_NAME_FIELD = new ParseField("index_name");
-    private static final ParseField IS_NATIVE_FIELD = new ParseField("is_native");
-    private static final ParseField LANGUAGE_FIELD = new ParseField("language");
-
-    private static final ParseField NAME_FIELD = new ParseField("name");
-    private static final ParseField PIPELINE_FIELD = new ParseField("pipeline");
-    private static final ParseField SCHEDULING_FIELD = new ParseField("scheduling");
-    private static final ParseField SERVICE_TYPE_FIELD = new ParseField("service_type");
-    private static final ParseField STATUS_FIELD = new ParseField("status");
-    private static final ParseField SYNC_CURSOR_FIELD = new ParseField("sync_cursor");
-    private static final ParseField SYNC_NOW_FIELD = new ParseField("sync_now");
+    protected static final ParseField ID_FIELD = new ParseField("connector_id");
```

Comment:
Why the change to protected?

Reply:
In `ConnectorIndexService` I support sorting by:
```
.sort(Connector.ID_FIELD.getPreferredName(), SortOrder.ASC);
``` 
So I need to have access to this field from a different class. 

For `update_x` actions I would need access to most of the parse fields in the connector doc. I decided to change all of the to protected for convenience.

Reply:
Thanks for explaining! If it's used by other classes in the package, it's better to use package protected access (no access modifier).

`protected` should be used just for subclass access. It can be confusing if subclasses are not using it or the class is not meant to be subclassed.

Reply:
Thank you for explanation :) I understand and I'm changing the access modifier then